pkg/controller/controller.go

```diff
@@ -411,3 +415,16 @@ func (c *Controller) getEffectiveNamespace(namespaceFromEnvironment, namespaceFr
 
 	return namespace
 }
+
+// hasOwnership returns true if the controller is the "owner" of the postgresql.
+// Whether it's owner is determined by the value of 'acid.zalan.do/controller'
+// annotation. If the value matches the controllerID then it owns it, or if the
+// controllerID is "" and there's no annotation set.
+func (c *Controller) hasOwnership(postgresql *acidv1.Postgresql) bool {
+	if postgresql.Annotations != nil {
+		if owner, ok := postgresql.Annotations[constants.PostgresqlControllerAnnotationKey]; ok {
+			return owner == c.controllerID
+		}
+	}
+	return c.controllerID == ""
```

Comment:
1. so the operator with `return c.controllerID == ""` will also be responsible for all PG clusters with non-empty `PostgresqlControllerAnnotationKey` that are not managed by other operator ? For example for the orphan ones ?
2. because of its importance, this method needs a couple of unit tests

Reply:
1. There can indeed be orphaned Postres clusters if the operator with the corresponding controller ID is gone. The operator with `c.controllerID == ""` will not take over. Not sure, if he should.
2. definitely. Will add them.

Reply:
please add unit tests to avoid regressions in the future